src/nodes.coffee

```diff
@@ -4551,6 +4554,7 @@ exports.For = class For extends While
     @awaitTag.error 'await must be used with for-from' if @await and not @from
     @range   = @source instanceof Value and @source.base instanceof Range and not @source.properties.length and not @from
     @pattern = @name instanceof Value
+    @name.unwrap().propagateLhs?(yes) if @pattern
```

Comment:
When there's a for variable "pattern" eg `for {a} in b`, it needs to know that it's a "left-hand-side" so that it can eg generate `ObjectPattern` AST nodes